pkg/unittest/valueutils/valueutils.go

```diff
@@ -4,12 +4,35 @@ import (
 	"bytes"
 	"fmt"
 	"io"
+	"regexp"
 	"strconv"
+	"strings"
 
 	"github.com/helm-unittest/helm-unittest/internal/common"
 	"github.com/vmware-labs/yaml-jsonpath/pkg/yamlpath"
 )
 
+// quotePath correctly quotes YAML Path keys with spaces while preserving valid structure
+func quotePath(path string) string {
+	if path == "" {
+		return ""
+	}
+
+	parts := strings.Split(path, ".")
+
+	for i, part := range parts {
+		if strings.Contains(part, " ") {
+			parts[i] = `["` + part + `"]`
+		}
+	}
+
+	quotedPath := strings.TrimPrefix(strings.Join(parts, "."), ".")
+	re := regexp.MustCompile(`\.(\["[^"]+"\])`)
```

Comment:
any chance to move this outside of the function? Is quite expensive to compile it on every function run

Example https://github.com/helm-unittest/helm-unittest/blob/b38a21fb4452700695ff6339338958a37ed6751c/pkg/unittest/test_suite.go#L25. Would be nice to have a description on what this regex should be doing